src/TypeScript.UnitTests/Rules/ActiveJavaScriptRulesProviderTests.cs

```diff
@@ -94,20 +98,82 @@ public void Get_RulesWithNullEslintKeysAreNotReturned()
         }
 
         [TestMethod]
-        public void Get_RulesWithConfigurations_ExpectedConfigsReturned()
+        public void Get_DefaultRulesConfig_RulesWithConfigurations_ExpectedConfigsReturned()
         {
             var config1 = new object();
             var config2 = "a default rule value";
 
             var ruleDefns = new RuleDefinitionsBuilder();
             ruleDefns.AddRule("aaa", configurations: new object[] { config1, config2 });
 
-            var testSubject = new ActiveJavaScriptRulesProvider(ruleDefns);
+            var testSubject = new ActiveJavaScriptRulesProvider(ruleDefns, EmptyUserSettingsProvider);
 
             var result = testSubject.Get().ToArray();
 
             CheckExpectedRuleKeys(result, "aaa");
-            result.First().Configurations.Should().ContainInOrder(config1, config2);
+            result[0].Configurations.Should().ContainInOrder(config1, config2);
+        }
+
+        [TestMethod]
+        [DataRow(true, true)]
+        [DataRow(false, false)]
+        [DataRow(false, true)]
+        [DataRow(true, false)]
+        public void Get_StandaloneMode_OverridesDefaultActivationLevel(
+            bool onByDefault, bool onInUserSettings)
+        {
+            // The value in the user settings should always override the default
+            var expectedRuleCount = onInUserSettings ? 1 : 0;
+
+            // Create definition and rule settings with a single rule
+            var ruleDefns = new RuleDefinitionsBuilder();
+            ruleDefns.AddRule(ruleKey: "javascript:rule1", activeByDefault: onByDefault, eslintKey: "esKey");
+
+            var userSettingsLevel = onInUserSettings ? RuleLevel.On : RuleLevel.Off;
+            var userSettings = new UserSettingsBuilder()
+                .Add("javascript:rule1", userSettingsLevel);
+
+            var testSubject = new ActiveJavaScriptRulesProvider(ruleDefns, userSettings);
+
+            var result = testSubject.Get().ToArray();
+
+            result.Length.Should().Be(expectedRuleCount);
+        }
+
+        [TestMethod]
+        public void Get_StandaloneMode_IrrelevantSettingsAreIgnored()
+        {
+            var ruleDefns = new RuleDefinitionsBuilder();
+            ruleDefns.AddRule(ruleKey: "javascript:activeRule1", activeByDefault: true, eslintKey: "active1");
+
+            var userSettings = new UserSettingsBuilder()
+                .Add("wrongLanguage:activeRule1", RuleLevel.Off) // Correct rule key but wrong repo -> ignored
+                .Add("javascript:unknownRule", RuleLevel.On); // Right repo, but rule key doesn't match a defined rule -> ignored
```

Comment:
Minor: I'll change the existing `RuleDefinitionsBuilder` helper to have a fluent API in a separate commit.